compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -115,52 +121,71 @@ LogicalResult distributeLocalMemory(ModuleOp moduleOp) {
         newDeallocOp->moveBefore(&newAllocOp->getBlock()->back());
         memrefToNew[allocOp] = newAllocOp;
       }
-      auto newAlloc = memrefToNew[allocOp];
-      rewriter.replaceAllUsesWith(subviewOp, newAlloc);
-      toBeErased.push_back(subviewOp);
     }
 
-    // Update the alloc's DMA users.
-    if (memrefToNew.contains(allocOp)) {
-      LLVM_DEBUG(llvm::dbgs()
-                 << "Update allocate DMA users: " << dmaUsers.size() << "\n");
-      auto newAlloc = memrefToNew[allocOp];
-      auto type = cast<MemRefType>(newAlloc.getType());
-      for (AMDAIE::DmaCpyNdOp dmaOp : dmaUsers) {
-        SmallVector<Value> empty;
-        rewriter.setInsertionPoint(dmaOp.getSourceObjectFifo());
-        auto source = rewriter.create<AMDAIE::LogicalObjectFifoFromMemrefOp>(
-            rewriter.getUnknownLoc(), LogicalObjectFifoType::get(type),
-            newAlloc.getResult());
-        rewriter.replaceOp(dmaOp.getSourceObjectFifo(), source);
-        rewriter.setInsertionPoint(dmaOp);
-        auto newDmaOp = rewriter.create<AMDAIE::DmaCpyNdOp>(
-            dmaOp.getLoc(), dmaOp.getTarget(), dmaOp.getTargetMixedOffsets(),
-            dmaOp.getTargetMixedSizes(), dmaOp.getTargetMixedStrides(), source,
-            dmaOp.getSourceMixedOffsets(), dmaOp.getSourceMixedSizes(),
-            dmaOp.getSourceMixedStrides());
-        rewriter.replaceOp(dmaOp, newDmaOp);
-        // We have to discard non-zero offsets as subview has been replaced by a
-        // dedicated allocated memref.
-        SmallVector<int64_t> allocShape(type.getShape());
-        (void)discardAllNonZeroOffsets<CopyOpOperateOn::Source>(
-            rewriter,
-            cast<AMDAIE::DoublyStridedOpInterface>(newDmaOp.getOperation()),
-            allocShape);
+    if (!memrefToNew.contains(allocOp)) return WalkResult::advance();
+
+    memref::AllocOp newAlloc = memrefToNew[allocOp];
+    for (Operation *userOp : allocOpUsers) {
+      if (auto subviewOp = dyn_cast<memref::SubViewOp>(userOp)) {
+        rewriter.replaceAllUsesWith(subviewOp, newAlloc);
+        toBeErased.push_back(subviewOp);
+      } else if (auto transferReadOp =
+                     dyn_cast<vector::TransferReadOp>(userOp)) {
+        rewriter.setInsertionPoint(transferReadOp);
+        Value transferRead = rewriter.create<vector::TransferReadOp>(
```

Comment:
Now, you have `transferReadOp` and `transferRead` and the difference between the two is unclear from the naming. So, I would call this `newTransferReadOp` instead.